x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -84,20 +80,36 @@ func (k msgServer) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake)
 	if balance.Amount.LT(msg.Amount) {
 		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidCoins, "balance is lower than redemption amount. redemption amount: %v, balance %v: ", msg.Amount, balance.Amount)
 	}
-	// UNBONDING RECORD KEEPING
-	userRedemptionRecord := recordstypes.UserRedemptionRecord{
-		Id:          redemptionId,
-		Sender:      senderAddr,
-		Receiver:    msg.Receiver,
-		Amount:      nativeAmount,
-		Denom:       hostZone.HostDenom,
-		HostZoneId:  hostZone.ChainId,
-		EpochNumber: epochTracker.EpochNumber,
-		// claimIsPending represents whether a redemption is currently being claimed,
-		// contingent on the host zone unbonding having status CLAIMABLE
-		ClaimIsPending: false,
+
+	// ----------------- UNBONDING RECORD KEEPING -----------------
+	// Fetch the record
+	senderAddr := sender.String()
+	redemptionId := recordstypes.UserRedemptionRecordKeyFormatter(hostZone.ChainId, epochTracker.EpochNumber, senderAddr)
+	userRedemptionRecord, userHasRedeemedThisEpoch := k.RecordsKeeper.GetUserRedemptionRecord(ctx, redemptionId)
+	if userHasRedeemedThisEpoch {
+		k.Logger(ctx).Info(fmt.Sprintf("UserRedemptionRecord found for %s", redemptionId))
+		// Add the unbonded amount to the UserRedemptionRecord
+		// The record is set below
+		userRedemptionRecord.Amount.Add(nativeAmount)
+	} else {
+		// First time a user is redeeming this epoch
+		userRedemptionRecord = recordstypes.UserRedemptionRecord{
+			Id:          redemptionId,
+			Sender:      senderAddr,
+			Receiver:    msg.Receiver,
+			Amount:      nativeAmount,
+			Denom:       hostZone.HostDenom,
+			HostZoneId:  hostZone.ChainId,
+			EpochNumber: epochTracker.EpochNumber,
+			// claimIsPending represents whether a redemption is currently being claimed,
+			// contingent on the host zone unbonding having status CLAIMABLE
+			ClaimIsPending: false,
+		}
+		k.Logger(ctx).Info(fmt.Sprintf("UserRedemptionRecord not found - creating for %s", redemptionId))
 	}
+
 	// then add undelegation amount to epoch unbonding records
+	// ASSUMPTION: we don't need to update the EpochUnbondingRecord directly (only the HostZoneUnbonding)
```

Comment:
is this meant to point out that the HZU's are pointers? Cause we do update the EUR [here](https://github.com/Stride-Labs/stride/pull/1009/files#diff-1d851c7e8c08659458b50e0840e7e3668f32b24ea7d0a4280adb13bba3b70379R153-R158)

Reply:
yeah this was just pointing out pointers, removed